js/RecentDonations.js

```diff
@@ -0,0 +1,196 @@
+(function() {
+var app = angular.module('RecentDonations', []);
+app.controller('RecentDonationsControl', function($scope) {
+
+    var urlVars = GLOBALS.getUrlVars();
+    var count = Number(urlVars['count']);
+    var uid = urlVars['id'];
+    var isTeam = urlVars['team'];
+
+    if (count === undefined) { count = 5 }
+
+    var contributions = []
+    var waitingOn = 0
+
+    var messageBeingRead = -1
+
+    $scope.contributions = []
+    $scope.errors = []
+
+    $scope.message = ""
+
+    // Use this to add an error to the stack to display.
+    var showError = function(problem, error) {
+      if (error) {
+        $scope.errors.push(problem + ": " + error)
+      } else {
+        $scope.errors.push(problem)
+      }
+      dataChanged()
+    }
+    
+    // Called whenever our data is updated by callbacks.
+    var dataChanged = function() {
+      // Only update once all our callbacks have added their data.
+      if (waitingOn == 0) {
+        
+      /* // Some test data.
+        contributions = [
+        {
+          name: "Alex Karantza",
+          date: Date.parse("07 05 1988"),
+          amount: "$100.00",
+          message: "I am the very model of a modern major general, Ive information vegetable animal and mineral; I know the kings of england and I quote the fights historical, from marathon to waterloo in order catagorical",
+        },
+        {
+          name: "Mike Stark",
+          date: Date.parse("08 01 1994"),
+          amount: "$99.00",
+          message: "DOUBLE HEAD",
+        },
+        {
+          name: "Adina Rubinoff",
+          date: Date.parse("10 18 1989"),
+          amount: "$101.00",
+          message: null,
+        },
+        ]
+        */
+
+        // Sort all our contributions
+        contributions.sort(function(a,b) { 
+          return b.date - a.date 
+        })
+    
+        var subset = contributions.slice(0, count)
+
+        // And send the latest 'count' to the UI.
+        //var subset = contributions.
+        $scope.contributions = subset
+        $scope.$apply()
+
+
+        // Now let's get a message from those contributors, if we can
+        if (subset.length > 0) {
+
+          // Starting after our current message, look and see if there's another message to display (wrapping back to the first if necessary)
+          for (var i = 1; i <= subset.length; i++) {
+            var idx = (messageBeingRead + i) % subset.length
+            var donation = subset[idx]
+            // If there is one, display it.
+            if (donation.message) { 
+              // Twitter rules.
+              var message = donation.message
+              if (message.length > 143) { 
```

Comment:
i think we should do a bigger cutoff than 143 (150, or maybe even 160). Just think what if the message was like 144 chars. You'd end with something like "You guys are all so gr..." instead of just showing "You guys are all so great!" I just think we should only truncate a significant amount of data. And in the end, 10 chars isn't a huge difference in width, but it is significant in # of words.